salt/pillar/__init__.py

```diff
@@ -302,6 +298,119 @@ def __init__(self, opts, grains, minion_id, saltenv, ext=None, functions=None,
             else:
                 log.error('Pillar data must be a dictionary')
 
+    @property
+    def pillar(self):
+        '''
+        Return the current pillar state.
+        '''
+        # FIXME: should this return a deepcopy rather than a reference?
+        # FIXME: should this merge in self.pillar_override?
+        # FIXME: maybe keep a hash of the accumulator to check if other objects need to regenerate?
+        pillar = copy.deepcopy(self._pillar_accumulator)
+        pillar = merge(
+            pillar,
+            self.pillar_override,
```

Comment:
Here the ``self.pillar_override`` is merged in *each* time ``self.pillar``.  The big question is how *persistent* should the override data be?  Does it simply seed the pillar rendering?  Can it be overwritten by a pillar provider during rendering?  Those questions still need to be answered so that I can close one of my ``FIXME``s.  Right now it is coded so that ``pillar_override`` is persistent and can't be overridden - it not only seeds the pillar rendering, but is also guaranteed to be in the final pillar structure.